recipes/pcs/hpc_ready_ami/assets/components/update-os.yaml

```diff
@@ -39,13 +39,16 @@ Resources:
         phases:
           - name: build
             steps:
-              - name: DownloadScript
-                action: ExecuteBash
+              - name: DownloadCommon
+                action: WebDownload
                 inputs:
-                  commands:
-                    - set -e
-                    - 'curl -fsSL https://${HpcRecipesS3Bucket}.s3.us-east-1.amazonaws.com/${HpcRecipesBranch}/recipes/pcs/hpc_ready_ami/assets/scripts/common.sh -o common.sh'
-                    - 'curl -fsSL https://${HpcRecipesS3Bucket}.s3.us-east-1.amazonaws.com/${HpcRecipesBranch}/recipes/pcs/hpc_ready_ami/assets/scripts/update-os.sh -o update-os.sh'
+                  - source: https://${HpcRecipesS3Bucket}.s3.us-east-1.amazonaws.com/${HpcRecipesBranch}/recipes/pcs/hpc_ready_ami/assets/scripts/common.sh
+                    destination: common.sh
+              - name: DownloadAgent
```

Comment:
Why not use:

```sh
  - name: UpdateLinux
    action: UpdateOS
```

Reply:
Does not work for [all OSs](https://docs.aws.amazon.com/imagebuilder/latest/userguide/toe-action-modules.html#action-modules-updateos) we want to support: "UpdateOS doesn't support Amazon Linux 2023 (AL2023)."